evaluators/rank/NdcgEvaluator/__init__.py

```diff
@@ -45,10 +45,18 @@ def evaluate(
             For detailed information, please check https://en.wikipedia.org/wiki/Discounted_cumulative_gain
         :return the evaluation metric value for the request document.
         """
+        # Information gain must be greater or equal to 0.
+        if any(item < 0 for item in actual) or any(item < 0 for item in desired):
+            raise ValueError('One or multiple score is less than 0.')
+        # Desired must in desc order.
+        if desired != sorted(desired, reverse=True):
+            raise ValueError('Information gain of desired must in a desc order.')
         actual_at_k = actual[:self.eval_at]
         desired_at_k = desired[:self.eval_at]
-        if len(actual) < 2:
-            raise ValueError(f'Expecting gains with minimal length of 2, {len(actual)} received.')
+        if len(actual_at_k) < 2:
```

Comment:
Why it has to be larger than 2? Is the formula not defined for that case?

Reply:
yes,  it's useless to check ndcg with one value, right? the score always equal to the gain itself.

Reply:
It is useless but not invalid, why make it an exception? To me exception is something really exceptional no?